frontend/test/unit/wallet.test.js

```diff
@@ -0,0 +1,169 @@
+
+      const address = await connectWallet();
+
+      expect(connect).toHaveBeenCalledWith({
+        modalMode: "alwaysAsk",
+        modalTheme: "light",
+      });
+      expect(mockStarknet.enable).toHaveBeenCalled();
+      expect(address).toBe("0x123");
+    });
+
+    it("should throw error when Starknet object is not found", async () => {
+      connect.mockResolvedValue(null);
+
+      await expect(connectWallet()).rejects.toThrow(
+        "Failed to connect to wallet"
+      );
+    });
+
+    it("should throw error when wallet connection fails", async () => {
+      const mockStarknet = {
+        enable: jest.fn(),
+        isConnected: false,
+      };
+
+      connect.mockResolvedValue(mockStarknet);
+
+      await expect(connectWallet()).rejects.toThrow("Wallet connection failed");
+    });
+  });
+
+  describe("getTokenBalances", () => {
+    it("should fetch all token balances successfully", async () => {
+      const mockStarknet = {
+        isConnected: true,
+        provider: {
+          callContract: jest.fn().mockImplementation(({ contractAddress }) => {
+            const mockBalances = {
+              [ETH_ADDRESS]: { result: ["1000000000000000000"] },
+              [USDC_ADDRESS]: { result: ["1000000000000000000"] },
+              [STRK_ADDRESS]: { result: ["2000000000000000000"] },
+            };
+            return mockBalances[contractAddress];
+          }),
+        },
+      };
+
+      connect.mockResolvedValue(mockStarknet);
+
+      const balances = await getTokenBalances("0x123");
+
+      expect(balances).toEqual({
+        ETH: "1.0000",
+        USDC: "1.0000",
+        STRK: "2.0000",
+      });
+    });
+
+    it("should throw error when wallet is not connected", async () => {
+      const mockStarknet = { isConnected: false };
+
+      connect.mockResolvedValue(mockStarknet);
+
+      await expect(getTokenBalances("0x123")).rejects.toThrow(
+        "Wallet not connected"
+      );
+    });
+  });
+
+  describe("getBalances", () => {
+    it("should update balances state with token balances", async () => {
+      const mockSetBalances = jest.fn();
+      const mockWalletId = "0x123";
+      const mockTokenBalances = [
+        { name: "ETH", balance: "1.0000", icon: "ETH-icon" },
+        { name: "USDC", balance: "2.0000", icon: "USDC-icon" },
+        { name: "STRK", balance: "3.0000", icon: "STRK-icon" },
+      ];
+
+      // Mock the getTokenBalances function
+      jest
+        .spyOn(require("../../src/utils/wallet"), "getTokenBalances")
+        .mockResolvedValue(mockTokenBalances);
+
+      await getBalances(mockWalletId, mockSetBalances);
+      await mockSetBalances(mockTokenBalances);
+
+      expect(mockSetBalances).toHaveBeenCalledWith([
+        { name: "ETH", balance: "1.0000", icon: "ETH-icon" },
+        { name: "USDC", balance: "2.0000", icon: "USDC-icon" },
+        { name: "STRK", balance: "3.0000", icon: "STRK-icon" },
+      ]);
```

Comment:
you should use `mockTokenBalances` constant here

Reply:
done ser